src/main/kotlin/uk/gov/dluhc/printapi/database/entity/AnonymousElectorDocument.kt

```diff
@@ -0,0 +1,134 @@
+package uk.gov.dluhc.printapi.database.entity
+
+import org.hibernate.Hibernate
+import org.hibernate.annotations.CreationTimestamp
+import org.hibernate.annotations.GenericGenerator
+import org.hibernate.annotations.Type
+import org.springframework.data.annotation.LastModifiedBy
+import org.springframework.data.jpa.domain.support.AuditingEntityListener
+import uk.gov.dluhc.printapi.database.repository.UUIDCharType
+import uk.gov.dluhc.printapi.database.repository.UseExistingOrGenerateUUID
+import java.time.Instant
+import java.time.LocalDate
+import java.util.UUID
+import javax.persistence.CascadeType
+import javax.persistence.Entity
+import javax.persistence.EntityListeners
+import javax.persistence.EnumType
+import javax.persistence.Enumerated
+import javax.persistence.FetchType
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.OneToMany
+import javax.persistence.OneToOne
+import javax.persistence.Table
+import javax.persistence.Version
+import javax.validation.constraints.NotNull
+import javax.validation.constraints.Size
+
+@Table
+@Entity
+@EntityListeners(AuditingEntityListener::class)
+class AnonymousElectorDocument(
+
+    @Id
+    @Type(type = UUIDCharType)
+    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "UUID")
+    @GenericGenerator(name = "UUID", strategy = UseExistingOrGenerateUUID.NAME)
+    var id: UUID? = null,
+
+    @field:NotNull
+    @field:Size(max = 20)
+    var certificateNumber: String? = null,
+
+    @field:NotNull
+    @Enumerated(EnumType.STRING)
+    var sourceType: SourceType? = null,
+
+    @field:NotNull
+    @field:Size(max = 255)
+    var sourceReference: String? = null,
+
+    @field:Size(max = 255)
+    var applicationReference: String? = null,
+
+    @field:NotNull
+    @Enumerated(EnumType.STRING)
+    var certificateLanguage: CertificateLanguage? = null,
+
+    @Enumerated(EnumType.STRING)
+    var supportingInformationFormat: SupportingInformationFormat? = null,
+
+    @field:NotNull
+    @field:Size(max = 255)
+    var photoLocationArn: String? = null,
+
+    @field:NotNull
+    @field:Size(max = 255)
+    var firstName: String? = null,
+
+    @field:Size(max = 255)
+    var middleNames: String? = null,
+
+    @field:NotNull
+    @field:Size(max = 255)
+    var surname: String? = null,
+
+    @OneToOne(cascade = [CascadeType.ALL])
+    var address: Address,
```

Comment:
I wonder if we should put the name fields and possibly the address within a separate object - e.g. `PersonalDetails`, or something like that? My understanding is that we need to keep this data for re-prints after the "application" data has been removed from VCA. I'm about to start refining the data retention side of things for AEDs, but I have a feeling we need to remove this personal data sooner (after 15 months?) than the data that's on the document itself. In which case, it will need to be nullable.

Reply:
As an embeddable or an `Entity` (similar to the `Delivery` that has separate table and entity)?
I considered both but wasn't sure modelling the tables/entities for data retention policy was a valid approach.  If we didn't have an existing `Address` Entity would have probably have modelled as an embeddable.

But given you're also questioning I think it's probably a valid approach - certainly the data is a logical unit 👍 Probably will call it Contact Details and will include phone number and email too 👍